lib/rubocop/cop/lint/no_rescue_error_message_checking.rb

```diff
@@ -0,0 +1,58 @@
+# typed: false
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Checks for the presence of error message checking within rescue blocks.
+      #
+      # @example
+      #
+      #   # bad
+      #   begin
+      #     something
+      #   rescue => e
+      #     if e.message.match?(/Duplicate entry/)
+      #       handle_error
+      #     end
+      #   end
+      #
+      #  # bad
+      #  begin
+      #    something
+      #  rescue => e
+      #    unless e.message.match?(/Duplicate entry/)
+      #      handle_error
+      #    end
+      #  end
+      #
+      #   # good
+      #   begin
+      #     something
+      #   rescue ActiveRecord::RecordNotUnique => e
+      #     handle_error
+      #   end
+      #
+      class NoRescueErrorMessageChecking < Base
+        MSG = 'Avoid checking error message while handling exceptions. ' \
+              'This is brittle and can break easily.'
+        METHODS_TO_CHECK = %i[match? include?].freeze
+
+        def on_rescue(node)
+          node.each_descendant(:if, :unless).each do |condition_node|
+            add_offense(condition_node) if message_check?(condition_node)
+          end
+        end
+
+        private
+
+        def message_check?(condition_node)
+          condition_node.condition&.type == :send &&
```

Comment:
It's preferable to use the built in node type matchers instead: 
```suggestion
          condition_node.condition.send_type? &&
```